src/core/utils/components/VirtualScroll.tsx

```diff
@@ -0,0 +1,232 @@
+  startIndex: number,
+  childrenLength: number,
+  scrollContainer: HTMLElement,
+) => {
+  return Math.min(
+    childrenLength - startIndex,
+    getCountOfNodesInHeight(childHeight, getElementHeight(scrollContainer)),
+  );
+};
+
+export type VirtualScrollProps = {
+  /**
+   * Length of the items to virtualize.
+   */
+  itemsLength: number;
+  /**
+   * Single item render function, which gives index of the item (0 based) in the data array
+   * and expects to get the JSX of that element to render.
+   * Recommended to memoize the reference of the function.
+   */
+  itemRenderer: (index: number) => JSX.Element;
+  /**
+   * Number of items to be rendered at the end.
+   * Not recommended to go lower than the visible items in viewport.
+   * @default 20
+   */
+  renderAdditional?: number;
+};
+
+/**
+ * `VirtualScroll` component to render only visible items + some additional out of a big list.
+ * Good use in expected big number of data either in some of the components (e.g. Table) or just simple list of DOM elements.
+ * It has 2 wrapper elements, so expect DOM tree change. One is used for setting full expected height in the scrollable container
+ * and other is for transformation (translateY) to show the correct part of the list.
+ * Currently it works only when items direct parent is a scrollable container (body scroll is not supported yet).
+ * It supports only static (same) height rows virtualization. Expect some issues, if list consists of different height elements.
+ * @example
+ * const itemRenderer = React.useCallback(() => (
+ *  <div key={index}>
+ *    This is my item #{index}
+ *  </div>
+ * ), [])
+ * <VirtualScroll
+ *  itemsLength={1000}
+ *  itemRenderer={itemRenderer}
+ * />
+ */
+export const VirtualScroll = ({
+  itemsLength: childrenLength,
+  itemRenderer,
+  renderAdditional = 20,
+}: VirtualScrollProps) => {
+  const [startNode, setStartNode] = React.useState(0);
+  const [visibleNodeCount, setVisibleNodeCount] = React.useState(0);
+  const scrollContainer = React.useRef<HTMLElement>();
+  const childrenParentRef = React.useRef<HTMLDivElement>(null);
+  const childHeight = React.useRef(0);
+  const [translateY, setTranslateY] = React.useState(0);
+  const onScrollRef = React.useRef<(e: Event) => void>();
+
+  // Find scrollable parent
+  // Needed only on init
+  React.useLayoutEffect(() => {
+    const scrollableParent = getScrollableParent(childrenParentRef.current);
+    scrollContainer.current = scrollableParent;
+  }, []);
+
+  const visibleChildren = React.useMemo(() => {
+    const arr = [];
+    const endIndex = Math.min(
+      childrenLength,
+      startNode + visibleNodeCount + renderAdditional,
+    );
+    for (let i = startNode; i < endIndex; i++) {
+      arr.push(itemRenderer(i));
+    }
+    return arr;
+  }, [
+    childrenLength,
+    itemRenderer,
+    renderAdditional,
+    startNode,
+    visibleNodeCount,
+  ]);
+
+  // Get child height when children available
+  React.useLayoutEffect(() => {
```

Comment:
I think we talked that with static height this is not needed anymore. But if you want to save it for dynamic height implementation maybe just create a branch out of this or PR will have history.

Reply:
what is not needed? So instead have param to pass height?

Reply:
This huge check is not needed as you just need to take first child and get its height.

Reply:
taking only first child